great_expectations/rule_based_profiler/config/base.py

```diff
@@ -0,0 +1,282 @@
+import copy
+import logging
+from typing import Any, Dict, List, Optional
+
+from great_expectations.data_context.types.base import BaseYamlConfig
+from great_expectations.marshmallow__shade import INCLUDE, Schema, fields, post_load
+from great_expectations.marshmallow__shade.decorators import post_dump
+from great_expectations.types import DictDot
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+class DomainBuilderConfig(DictDot):
+    def __init__(
+        self,
+        class_name: str,
+        module_name: Optional[str] = None,
+        batch_request: Optional[Dict[str, Any]] = None,
+        **kwargs,
+    ):
+        self._class_name = class_name
+        self._module_name = module_name
+        self.batch_request = batch_request
+        for k, v in kwargs.items():
+            setattr(self, k, v)
+            logger.info(
+                'Set "self.%s = %s" during DomainBuilderConfig instantiation', k, v
+            )
+
+    @property
+    def class_name(self) -> str:
+        return self._class_name
+
+    @property
+    def module_name(self) -> Optional[str]:
+        return self._module_name
+
+
+class DomainBuilderConfigSchema(Schema):
+    class Meta:
+        unknown = INCLUDE
+
+    class_name = fields.String(required=True)
+    module_name = fields.String(
+        required=False,
+        all_none=True,
+        missing="great_expectations.rule_based_profiler.domain_builder",
+    )
+    batch_request = fields.Dict(keys=fields.String(), required=False, allow_none=True)
+
+    @post_load
+    def make_config(self, data: dict, **kwargs) -> DomainBuilderConfig:
+        return DomainBuilderConfig(**data)
+
+    @post_dump
+    def remove_nulls(self, data: dict, **kwargs) -> dict:
+        res = copy.deepcopy(data)
+        for k, v in data.items():
+            if v is None:
+                res.pop(k)
+                logger.info("Removed '%s' due to null value", k)
+        return res
```

Comment:
This pattern tends to repeat between Schemas; should we perhaps think about extending `Schema` and subclassing all RBP config classes?

Reply:
@cdkini Please see the above comment?  Perhaps a quick gain with little effort all around?

Reply:
I think we still have the overarching issue of repeating code (`post_load` and `post_dump`) for each schema class. Perhaps a minor issue since code duplication is not the worst thing in the world.

Reply:
but .... code duplication IS the worst thing in the world!

Reply:
Moved to base class; resolving